### DISCUSSION TITLE
Upgrade `mutt_str_append_item()` to use `struct Buffer`

### DISCUSSION THREAD

**Discussion by flatcap** [neomutt/neomutt#4129] [Development]
`mutt_str_append_item()` appends a string to another string, with a separator character.
This means resizing the original string.

This function would be simpler if it used `struct Buffer`.
Its callers could then use `buf_pool_get()` and `buf_pool_release()`.

Most callers use the string and discard it.
The two marked with a **red dot** need a copy.

<img src="https://github.com/neomutt/test-gfx/raw/main/2023-12-09_mutt_str_append_item/append.svg">

Source: [gv](https://github.com/neomutt/test-gfx/raw/main/2023-12-09_mutt_str_append_item/append.gv) [svg](https://github.com/neomutt/test-gfx/raw/main/2023-12-09_mutt_str_append_item/append.svg)

**Comment by roccoblues** [comment]
I started this here: https://github.com/neomutt/neomutt/pull/4159

**Comment by flatcap** [comment]
Done: eb406060e
